packages/astro/src/default/components/MobileContentToggle.astro

```diff
@@ -0,0 +1,50 @@
+---
+
+---
+
+<div class="h-12 sm:hidden"></div>
+<view-toggle
+  class="fixed sm:hidden h-12 bottom-0 w-full bg-tk-elements-app-backgroundColor border-t border-tk-elements-app-borderColor flex justify-center items-center z-60 text-tk-elements-app-textColor"
+>
+  <button aria-hidden="true">Tutorial</button>
+  <button aria-pressed="true" class="rounded-full w-8 h-4 p-0.5 bg-gray-4 dark:bg-gray-6 mx-2 relative">
+    <span class="inline-block transition-all absolute top-0.5 left-0.5 rounded-full bg-white dark:bg-gray-2 w-3 h-3"
+    ></span>
+  </button>
+  <button aria-hidden="true">Editor</button>
+</view-toggle>
+<script>
+  import { viewStore, type View } from '../stores/view-store';
+
+  class ViewToggle extends HTMLElement {
+    constructor() {
+      super();
+
+      const [tutorialBtn, toggleButton, editorButton] = this.querySelectorAll(
+        ':scope > button',
+      ) as NodeListOf<HTMLButtonElement>;
```

Comment:
Tip: `querySelectorAll` has good type narrowing. Instead of using type assertion on it's output, we could use type assertion on the input. It's much better to say `':scope > button'` selector captures `buttons`, than to say that `querySelectorAll` returns `NodeListOf<HTMLButtonElement>`.

```suggestion
      const [tutorialBtn, toggleButton, editorButton] = this.querySelectorAll(
        ':scope > button' as 'button',
      );
```

Reply:
Oh nice! I didn't knew about that trick, really cool :smiley:

Reply:
TIL!